test/test_torch.py

```diff
@@ -6977,15 +6977,64 @@ def compare_strides(s1, s2, div):
 
     @skipMeta
     @dtypes(*torch.testing.get_all_dtypes())
-    def test_dlpack_conversion(self, device, dtype):
-        # DLpack does not explicitly support bool
+    def test_dlpack_capsule_conversion(self, device, dtype):
+        # DLpack does not explicitly support bool (xref dmlc/dlpack#75)
         # It does it through uint8 type
-        if dtype is torch.bool:
+        if dtype is torch.bool or 'xla' in device:
             return
         x = make_tensor((5,), device, dtype, low=-9, high=9)
         z = from_dlpack(to_dlpack(x))
         self.assertEqual(z, x)
 
+    @skipMeta
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_dlpack_protocol_conversion(self, device, dtype):
```

Comment:
We should add a test that compares the DLPack semantics with our .numpy() and from_numpy() semantics. If/when NumPy implements the protocol we could really validate that the behavior is the same

Reply:
Sorry I don't understand what should be compared here.

Reply:
Now, I think I understood it,
A test that checks that tensors from dlpack can't be resized, or tensors with gradients can't be exported in the same sense that numpy does, is this correct?

Reply:
Yes: tensors with gradients or the conjugate bit set, I think, can't be exported.

For importing we should verify that the underlying memory is shared by writing to it on both CPU and CUDA.

NumPy arrays can also be non-writable, which we check for on import. I'm not sure what (if any) special properties DLPack capsules have that PyTorch can't emulate.